File: performance-os/app/models/feedback.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey, Boolean
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import Base

class Feedback(Base):
    __tablename__ = "feedbacks"
    
    id = Column(Integer, primary_key=True, index=True)
    organization_id = Column(Integer, ForeignKey("organizations.id"), nullable=False, index=True)
    
    sender_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    receiver_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    
    feedback_type = Column(String, nullable=False) # "kudo", "constructive"
    message = Column(Text, nullable=False)
    is_public = Column(Boolean, default=True)
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    
    # Relationships
    sender = relationship("User", foreign_keys=[sender_id])
    receiver = relationship("User", foreign_keys=[receiver_id])

File: performance-os/frontend/src/types/Feedback.ts
export interface Feedback {
    id: number;
    organization_id: number;
    sender_id: number;
    receiver_id: number;
    feedback_type: string;
    message: string;
    is_public: boolean;
    created_at: string;
}

export interface UserResponse {
    id: number;
    email: string;
    full_name: string;
    job_title: string;
    department: string | null;
    role: string;
    organization_id: number;
    manager_id: number | null;
    is_active: boolean;
    created_at: string;
}
